xonsh/tools.py

```diff
@@ -1634,13 +1654,45 @@ def lazyget(self, key, default=None):
         """A lazy value getter."""
         return self._cmds_cache.get(key, default)
 
+    def locate_binary(self, name):
```

Comment:
If you move the stuff about the current directory to `lazy_locate_binary()`  and then here you just do

```
def locate_binary(self, name):
    cmd = self.all_commands
    return self.lazy_locate_binary(name)
```